scripts/kconfig/kconfigfunctions.py

```diff
@@ -193,6 +193,34 @@ def _node_int_prop(node, prop, unit=None):
     return node.props[prop].val >> _dt_units_to_scale(unit)
 
 
+def _node_array_prop(node, prop, index=0, unit=None):
+    """
+    This function takes a 'node' and  will look to see if that 'node' has a
+    property called 'prop' and if that 'prop' is an array type will return
+    the value of the property 'prop' at the given 'index' as either a string int
+    or string hex value. If the property 'prop' is not found or the given 'index'
+    is out of range it will return 0.
+
+    The function will divide the value based on 'unit':
+        None        No division
+        'k' or 'K'  divide by 1024 (1 << 10)
+        'm' or 'M'  divide by 1,048,576 (1 << 20)
+        'g' or 'G'  divide by 1,073,741,824 (1 << 30)
+    """
+    if not node:
+        return 0
+
+    if prop not in node.props:
+        return 0
+    if node.props[prop].type != "array":
+        print("Error, not error prop")
```

Comment:
How will Kconfig behave if we print errors here ?

I assume bad, and in case such error occur, then how is the end-user expected to take action ?

Reply:
From what I have seen, the prints are added to the compile log. Other than that, Kconfig behaves the same as if the prints are not present. Anyhow, I will remove the prints as they are not used in other functions in this file and were primarily included for debugging. 

Regarding how the end-user is expected to take action, I followed the same pattern of returning 0 on error as in `_node_int_prop`. So if there is a mismatch between the device tree and the Kconfig file, the end-user will see 0 as the Kconfig value. It might be an idea to add compile warnings if this happens, but I don't think that should be in this PR as it requires updates for more functions in this file.